File: cucumber/src/main/java/com/cucumber/framework/pages/InventoryPage.java
package com.cucumber.framework.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * InventoryPage — SauceDemo product catalog and cart operations.
 */
public class InventoryPage extends BasePage {

    private final By cartBadge = By.className("shopping_cart_badge");
    private final By cartLink  = By.className("shopping_cart_link");

    public InventoryPage(WebDriver driver) {
        super(driver);
    }

    public void addItemToCart(String itemSlug) {
        click(By.id("add-to-cart-" + itemSlug));
    }

    public void removeItemFromCart(String itemSlug) {
        click(By.id("remove-" + itemSlug));
    }

    public void goToCart() {
        click(cartLink);
        waitForUrl("cart");
    }

    public String getCartBadgeCount() {
        return getText(cartBadge);
    }

    public boolean isCartBadgeVisible() {
        return isVisible(cartBadge);
    }
}

File: playwright-dotnet/tests/playwright-ts/pages/inventoryPage.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './basePage';

/**
 * InventoryPage — wraps SauceDemo product listing interactions.
 *
 * Locator strategy:
 *   - Cart link: getByTestId — data-test='shopping-cart-link'
 *   - Cart badge: scoped CSS on the cart link — no data-test attribute available
 *   - Add/Remove buttons: getByTestId with dynamic slug — data-test='add-to-cart-{slug}'
 *
 * C# equivalent: InventoryPage.cs
 */
export class InventoryPage extends BasePage {
  readonly cartLink: Locator;
  readonly cartBadge: Locator;

  constructor(page: Page) {
    super(page);
    this.cartLink = page.getByTestId('shopping-cart-link');
    this.cartBadge = page.locator('.shopping_cart_badge');
  }

  async addItemToCart(itemSlug: string): Promise<void> {
    await this.page.getByTestId(`add-to-cart-${itemSlug}`).click();
  }

  async removeItemFromCart(itemSlug: string): Promise<void> {
    await this.page.getByTestId(`remove-${itemSlug}`).click();
  }

  async goToCart(): Promise<void> {
    await this.cartLink.click();
  }

  async getCartBadgeCount(): Promise<string> {
    return await this.cartBadge.innerText();
  }

  async isCartBadgeVisible(): Promise<boolean> {
    return await this.cartBadge.isVisible();
  }
}
